src/Frame.cpp

```diff
@@ -1,227 +1,493 @@
+      auto i_radius = guess_bonds_radius(topology_[i]);
+      if (!i_radius) {
+         throw error(
+            "missing Van der Waals radius for '{}'", topology_[i].type()
+         );
+      }
+      for (size_t j = i + 1; j < size(); j++) {
+         auto j_radius = guess_bonds_radius(topology_[j]);
+         if (!j_radius) {
             throw error(
-                "missing Van der Waals radius for '{}'", topology_[i].type()
+               "missing Van der Waals radius for '{}'", topology_[j].type()
             );
-        }
-        for (size_t j = i + 1; j < size(); j++) {
-            auto j_radius = guess_bonds_radius(topology_[j]);
-            if (!j_radius) {
-                throw error(
-                    "missing Van der Waals radius for '{}'", topology_[j].type()
-                );
-            }
-            auto d = distance(i, j);
-            auto radii = i_radius.value() + j_radius.value();
-            if (0.03 < d && d < 0.6 * radii && d < cutoff) {
-                topology_.add_bond(i, j);
-            }
-        }
-    }
-
-    auto bonds = topology().bonds();
-    auto to_remove = std::vector<Bond>();
-    // We need to remove bonds between hydrogen atoms which are bonded more than
-    // once
-    for (auto& bond : bonds) {
-        auto i = bond[0], j = bond[1];
-        if (topology_[i].type() != "H") {
-            continue;
-        }
-        if (topology_[j].type() != "H") {
-            continue;
-        }
-
-        auto nbonds = std::count_if(
-            std::begin(bonds), std::end(bonds), [=](const Bond& b) {
-                return b[0] == i || b[0] == j || b[1] == i || b[1] == j;
-            });
-        assert(nbonds >= 1);
-
-        if (nbonds != 1) {
-            to_remove.push_back(bond);
-        }
-    }
-
-    for (auto& bond : to_remove) {
-        topology_.remove_bond(bond[0], bond[1]);
-    }
+         }
+         auto d = distance(i, j);
+         auto radii = i_radius.value() + j_radius.value();
+         if (0.03 < d && d < 0.6 * radii && d < cutoff) {
+            topology_.add_bond(i, j);
+         }
+      }
+   }
+
+   auto bonds = topology().bonds();
+   auto to_remove = std::vector<Bond>();
+   // We need to remove bonds between hydrogen atoms which are bonded more than
+   // once
+   for (auto& bond : bonds) {
+      auto i = bond[0], j = bond[1];
+      if (topology_[i].type() != "H") {
+         continue;
+      }
+      if (topology_[j].type() != "H") {
+         continue;
+      }
+
+      auto nbonds = std::count_if(
+         std::begin(bonds), std::end(bonds), [=](const Bond& b) {
+            return b[0] == i || b[0] == j || b[1] == i || b[1] == j;
+         });
+      assert(nbonds >= 1);
+
+      if (nbonds != 1) {
+         to_remove.push_back(bond);
+      }
+   }
+
+   for (auto& bond : to_remove) {
+      topology_.remove_bond(bond[0], bond[1]);
+   }
+}
+
+void Frame::getMinMaxBox(std::vector<Vector3D> pos, int n, Vector3D& min, Vector3D& max) {
```

Comment:
I absolutely missed that one, thank you for pointing out! 
About possible skewed triclinic cells, this function compute the dimensions of a cubic box which contains all the atoms of the structure by using atoms located at the extremities of each dimension. Therefore, I can't see how a different cell type will make it not work.

Reply:
Actually, I could not pinpoint issues specific to skewed cells, but there are a few issues related to periodicity:

![cell-list-ortho](https://user-images.githubusercontent.com/6141587/99809578-c6518800-2b42-11eb-9547-47a0a7867adf.png)

Here the unit cell is in black, and images in grey. The boxes created by the cell list (I'll call them boxes to prevent confusion with the unit cell) are in red. Atoms are in bright colors, and atom images semi-transparent.

One issue with the current implementation is that it will not find the bond between atoms far from one another in the central cell image (red and green atoms), although there should be one. A second related issue is the one of atoms outside the central image, that should have bond with atoms inside the central image (orange and blue).

I'll continue looking around about this skewed cell question.

Reply:
Thanks for the illustration. Indeed, I can clearly see the issue regarding periodicity. But I am wondering why the unit cell (in black) doesn't superimpose to the boxes created by the cell list (in red) which is containing all the atoms of the structure. Is it because of the chemfiles' way of defining a unit cell in periodic boundary conditions?

Reply:
>  I am wondering why the unit cell (in black) doesn't superimpose to the boxes created by the cell list (in red) 

Isn't the code in this PR using the min/max positions of atoms in the system to create the cell list? I agree that it should use the unit cell faces instead.

Reply:
Yes it is using min/max positions to define the box containing the structure. I will check how to use the unit cell faces instead in order to better manage periodicity.